bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/DaikinWebTargets.java

```diff
@@ -183,73 +182,72 @@ public void setAirbaseZoneInfo(AirbaseZoneInfo zoneinfo) throws DaikinCommunicat
     }
 
     private String invoke(String uri) throws DaikinCommunicationException {
-        return invoke(uri, new HashMap<>());
-    }
-
-    private String invoke(String uri, Map<String, String> params) throws DaikinCommunicationException {
-        String uriWithParams = uri + paramsToQueryString(params);
-        logger.debug("Calling url: {}", uriWithParams);
-        String response;
-        synchronized (this) {
-            try {
-                if (httpClient != null) {
-                    response = executeUrl(uriWithParams);
-                } else {
-                    // a fall back method
-                    logger.debug("Using HttpUtil fall scback");
-                    response = HttpUtil.executeUrl("GET", uriWithParams, TIMEOUT_MS);
+        return invoke(uri, null);
+    }
+
+    private synchronized String invoke(String url, @Nullable Map<String, String> params)
+            throws DaikinCommunicationException {
+        int attemptCount = 1;
+        try {
+            while (true) {
+                try {
+                    String result = executeUrl(url, params);
+                    if (attemptCount > 1) {
+                        logger.debug("HTTP request successful on attempt #{}: {}", attemptCount, url);
+                    }
+                    return result;
+                } catch (ExecutionException | TimeoutException e) {
```

Comment:
You do not catch  DaikinCommunicationException to reyry?

Reply:
No, it's caught by the caller of invoke, up the call chain. Note the invoke method throws it.

Reply:
executeUrl can throw 
a DaikinCommunicationForbiddenException. As you do not catch it, you will not retry the call in this case.
Is it expected? 
I imagine you will get the same 403 error if you retry. So it could make sense. 
Please just confirm,

Reply:
> executeUrl can throw
> a DaikinCommunicationForbiddenException. As you do not catch it, you will not retry the call in this case.
> Is it expected?

Correct. The DaikinCommunicationForbiddenException is not an error due to communication glitch, therefore there is no point in retrying. It happened because the controller requires authentication. The caller upstream will raise an error message telling the user to check the UUID / key that needs to be set in order to fulfil the authentication requirement.

Reply:
This is where it is handled: https://github.com/openhab/openhab-addons/pull/12239/files#diff-9cde2c78230ff5c6210b31fab607edd2ef651f49fb55f829ec8d4fd58412ba83L187

Reply:
@lolodomo